girder_annotation/girder_large_image_annotation/models/annotation.py

```diff
@@ -450,6 +450,46 @@ class AnnotationSchema:
         ]
     }
 
+    overlaySchema = {
+        '$schema': 'http://json-schema.org/schema#',
+        'type': 'object',
+        'properties': {
+            'girderId': {
+                'type': 'string',
+                'description': 'Girder item ID containing the image to '
+                               'overlay.'
+            },
+            'opacity': {
+                'type': 'number',
+                'minimum': 0,
+                'maximum': 1,
+                'description': 'Default opacity for this image overlay. Must '
+                               'be between 0 and 1. Defaults to 1.'
+            },
+            'location': {
+                'type': 'object',
+                'description': 'Position (upper-left coordinate, width, and '
+                               'height) of the overlay.',
+                'properties': {
+                    'upperLeftCorner': coordSchema,
+                    'width': {
+                        'type': 'number',
+                        'minimum': 0
```

Comment:
We probably will eventually want to allow for an affine transform matrix instead.  We end up with six values: xoffset, yoffset, plus a 2x2 scaling matrix [[s11, s12], [s21, s22]].  For the width/height condition, this ends up being [[specified width/image width, 0], [0, specified height/image height]].  In many instances, we don't want to apply a scale or skew, but just an offset and a rotation, but we can leave that to whoever specifies the matrix.

Reply:
We could add an optional field of `transform` which would be an array of exactly four numbers (and document them).  In this case, it would be nice to specify that you can't specify a transform AND a width or height.

Reply:
See 95c8b45. I think if we eventually want to allow for an affine transform matrix, it makes sense to be able to specify that now.